Sort Array of 0's 1's 2's in C++

HariniJeyaraman opened the pull request:
<!-- Add the issue number that is fixed by this PR (In the form Fixes #45) -->
Fixes #1161

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-  Added implementation of Sort in array containing only 0's 1's and 2's in C++
- 

#### Languages Used:
-  C++
- 

#### Files Added:
-  Sort_Array_zero_one_two.cpp
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

HariniJeyaraman commented:
@jainaman224 @SakshiShreya please review the code . Implemented in C++ to segregate 0's 1's and 2's in the array.

jainaman224 commented:
Complete PR template. Issue number is missing.

HariniJeyaraman commented:
@jainaman224 @SakshiShreya  updated the PR template. Filled the issue number, file name.

jainaman224 commented on Sort_Array_zero_one_two/segregate_zero_one_two.cpp:
```diff
@@ -0,0 +1,51 @@
+/*Given an array A[] consisting 0s, 1s and 2s. The task is to write a function that sorts the given array. 
```
Fix file and folder name.

jainaman224 commented on Sort_Array_zero_one_two/segregate_zero_one_two.cpp:
```diff
@@ -0,0 +1,51 @@
+
+void sort_zero_one_two(int a[],int n)
+{
+    int zero=0; //counter to store number of 0's
```
Add spaces around operators like `+`, `=`.

jainaman224 commented on Sort_Array_zero_one_two/segregate_zero_one_two.cpp:
```diff
@@ -0,0 +1,51 @@
+Enter the number of elements of array: 7                                                                                                                                                                                                                  
+Enter the array elements: 1 0 2 1 1 0 2                                                                                                                                                                                                             
+Original array : 1 0 2 1 1 0 2 
+Sorted array   : 0 0 1 1 1 2 2  */
```
Add single new line at the end of file.

jainaman224 commented on Sort_Array_zero_one_two/segregate_zero_one_two.cpp:
```diff
@@ -0,0 +1,51 @@
+    cout<<"\nSorted array   : ";
+    for(int i=0;i<n;i++)
+        cout<<a[i]<<" ";
+    cout<<"\n";
```
Use endl.

HariniJeyaraman commented:
What should be the file and folder name

On Sun, 8 Mar 2020, 01:12 Aman Jain, <notifications@github.com> wrote:

> *@jainaman224* requested changes on this pull request.
> ------------------------------
>
> In Sort_Array_zero_one_two/segregate_zero_one_two.cpp
> <https://github.com/jainaman224/Algo_Ds_Notes/pull/1999#discussion_r389306821>
> :
>
> > @@ -0,0 +1,51 @@
> +/*Given an array A[] consisting 0s, 1s and 2s. The task is to write a function that sorts the given array.
>
> Fix file and folder name.
> ------------------------------
>
> In Sort_Array_zero_one_two/segregate_zero_one_two.cpp
> <https://github.com/jainaman224/Algo_Ds_Notes/pull/1999#discussion_r389306832>
> :
>
> > @@ -0,0 +1,51 @@
> +/*Given an array A[] consisting 0s, 1s and 2s. The task is to write a function that sorts the given array.
> +The function puts all 0s first, then all 1s and all 2s in last.*/
> +
> +#include<stdlib.h>
> +#include<iostream>
> +using namespace std;
> +
> +void sort_zero_one_two(int a[],int n)
> +{
> +    int zero=0; //counter to store number of 0's
>
> Add spaces around operators like +, =.
> ------------------------------
>
> In Sort_Array_zero_one_two/segregate_zero_one_two.cpp
> <https://github.com/jainaman224/Algo_Ds_Notes/pull/1999#discussion_r389306850>
> :
>
> > +    cout<<"Original array : ";
> +    for(int i=0;i<n;i++)
> +        cout<<a[i]<<" ";
> +    sort_zero_one_two(a,n); //calling the sort function
> +    cout<<"\nSorted array   : ";
> +    for(int i=0;i<n;i++)
> +        cout<<a[i]<<" ";
> +    cout<<"\n";
> +    return 0;
> +}
> +
> +/* Sample Input - Output
> +Enter the number of elements of array: 7
> +Enter the array elements: 1 0 2 1 1 0 2
> +Original array : 1 0 2 1 1 0 2
> +Sorted array   : 0 0 1 1 1 2 2  */
>
> Add single new line at the end of file.
> ------------------------------
>
> In Sort_Array_zero_one_two/segregate_zero_one_two.cpp
> <https://github.com/jainaman224/Algo_Ds_Notes/pull/1999#discussion_r389306861>
> :
>
> > +{
> +    int n;
> +    cout<<"Enter the number of elements of array: ";
> +    cin>>n;
> +    int a[n];
> +    cout<<"Enter the array elements: ";
> +    for(int i=0;i<n;i++)
> +        cin>>a[i];
> +    cout<<"Original array : ";
> +    for(int i=0;i<n;i++)
> +        cout<<a[i]<<" ";
> +    sort_zero_one_two(a,n); //calling the sort function
> +    cout<<"\nSorted array   : ";
> +    for(int i=0;i<n;i++)
> +        cout<<a[i]<<" ";
> +    cout<<"\n";
>
> Use endl.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jainaman224/Algo_Ds_Notes/pull/1999?email_source=notifications&email_token=AN5R2RMCQCKZUGWFGSSDEWTRGKPUDA5CNFSM4LALCADKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCYM3WSQ#pullrequestreview-370785098>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN5R2ROINTQABPMHJKFL5Y3RGKPUDANCNFSM4LALCADA>
> .
>

HariniJeyaraman commented:
Fixed the file and folder name, added spaces around operators, added blank link at end of file, used endl instead of \n. @jainaman224  please review again.
